tests/data_io/test_csv.py

```diff
@@ -284,3 +284,75 @@ def test_bad_encoding():
     filepath = os.path.join(get_toolbox_root_dir(), "tests/data_io/temp_bad_encoding.csv")
     csv.from_dataset(dataset, filepath)
     os.remove(filepath)
+
+
+@pytest.mark.parametrize(
+    "columns, column_name_dict",
+    [
+        (
+            None,
+            {"first_name": "new_first_name", "last_name": "family_name"},
+        ),  # basic test renaming test
+        (
+            None,
+            {"not_real_col": "full_name", "last_name": "family_name"},
+        ),  # test with column name not present in the data
+        (
+            ["tamr_id", "all_names", "ssn", "last_name", "first_name"],
+            {"last_name": "family_name"},
+        ),  # test with columns subset specified
```

Comment:
Can you add a case where there are multiple columns renamed, and the order in `columns` and `column_name_dict` conflict?  The result should be that the order in `columns` is obeyed.